### PR TITLE
Bug 1457945, Clarified default behavior in Garbage Collection docs

### PR DESCRIPTION
**Pull request by ahardin-rh**
https://bugzilla.redhat.com/show_bug.cgi?id=1457945

Preview Build:
http://file.rdu.redhat.com/~ahardin/Dec012017/garbage-collection-bz/admin_guide/garbage_collection.html

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@sjenning @sdodson PTAL

File: admin_guide/garbage_collection.adoc [hunk 1]
```diff
@@ -15,14 +15,20 @@ toc::[]
...
 
 [[container-garbage-collection]]
-
 == Container Garbage Collection
 
+Container garbage collection is enabled by default and happens automatically in
+response to eviction thresholds being reached. The node tries to keep any pod
+that is still accessible from the API. If the pod has been deleted, the
+containers will be as well. Containers are only deleted when the pod object is
+gone from {product-title}. If the node is under disk pressure, it will remove
```
Comment (sjenning) [inline admin_guide/garbage_collection.adoc]: I would say "Containers are preserved as long the pod is not deleted and there an eviction threshold is not reached."

File: admin_guide/garbage_collection.adoc [hunk 2]
```diff
@@ -32,14 +38,14 @@ The policy for container garbage collection is based on three node settings:
...
+default is *0*. Use *0* for no limit. Values for this setting can be
 specified using unit suffixes such as *h* for hour, *m* for minutes, *s* for seconds.
 
 |`*maximum-dead-containers-per-container*`
-|The number of instances to retain per pod container. The default is *2*.
+|The number of instances to retain per pod container. The default is *1*.
 
 |`*maximum-dead-containers*`
-|The maximum number of total dead containers in the node. The default is *240*.
+|The maximum number of total dead containers in the node. The default is *-1*.
```
Comment (sjenning) [inline admin_guide/garbage_collection.adoc]: Might indicated that -1 means unlimited

Also all 3 of these flags are deprecated.

Comment (propyless) [reply admin_guide/garbage_collection.adoc]: What are the new flags?

Comment (sjenning) [reply admin_guide/garbage_collection.adoc]: https://kubernetes.io/docs/tasks/administer-cluster/out-of-resource/#deprecation-of-existing-feature-flags-to-reclaim-disk

Comment (ahardin-rh) [reply admin_guide/garbage_collection.adoc]: @sjenning Hmm so if these are all deprecated and have no direct replacements according to that link you shared, I'm not sure what to do. It was indicated that the docs mostly need to have these values updated, but I'm not sure when the flags became deprecated and what versions of our docs they need to be removed from.

Comment (sjenning) [reply admin_guide/garbage_collection.adoc]: @ahardin-rh they are deprecated at least as far back as 3.6, possibly 3.5 as well.  @derekwaynecarr do you know when eviction deprecated these flags?

Comment (ahardin-rh) [reply admin_guide/garbage_collection.adoc]: @derekwaynecarr Any guidance here? I want to be sure that when we update the docs across versions, we're not including any outdated information, including deprecated flags. Thanks!

Comment (vikram-redhat) [reply admin_guide/garbage_collection.adoc]: @sjenning is there anyone else who can answer this other than @derekwaynecarr ?

Comment (ahardin-rh) [reply admin_guide/garbage_collection.adoc]: @sjenning @derekwaynecarr PTAL. I'm currently blocked and not sure what is needed here.

File: admin_guide/garbage_collection.adoc [hunk 3]
```diff
@@ -15,14 +15,21 @@ toc::[]
...
 * xref:image-garbage-collection[Image garbage collection]: Removes images not
-referenced by any running pods. Typically run every five minutes.
+referenced by any running pods.
 
 [[container-garbage-collection]]
-
 == Container Garbage Collection
 
+Container garbage collection is enabled by default and happens automatically in
+response to eviction thresholds being reached. The node tries to keep any pod
```
Comment (derekwaynecarr) [inline admin_guide/garbage_collection.adoc]: node tries to keep any container for any pod accessible from the API

File: admin_guide/garbage_collection.adoc (hunk 2 above)
Comment (derekwaynecarr) [reply admin_guide/garbage_collection.adoc]: the flags were deprecated in kube 1.4, but still remain through kube 1.10.

we should continue to document them as I do not see them being removed for the next 6 mos.

**Comment by derekwaynecarr** [comment]
one nit, LGTM

**Comment by ahardin-rh** [comment]
Thanks! Updated.

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../admin_guide/garbage_collection.adoc#admin-guide-garbage-collection[Garbage Collection]
|Added clarifying details about the default behavior of garbage collection.
%

**Pull request merged**